### PR TITLE
Make it possible to swap browsers

### PR DESCRIPTION
**Pull request by grdw**
Currently `curl` is the browser of choice, but it might be interesting to some to swap this browser out with something else.

### CODE REVIEW

File: lib/url_fetcher.rb [hunk 1]
```diff
@@ -6,15 +6,21 @@ class UrlFetcher
 
     class UrlFetcherError < StandardError; end
 
-    def self.fetch(url, attributes = [], cache_time = nil)
-      new(url, attributes, cache_time).fetch
+    def self.fetch(
+      url,
+      attributes = [],
+      wrapper = PatronBrowser.new,
```
Comment (lorenzograndi4) [inline lib/url_fetcher.rb:12]: I'm confused, is this just the default value for the `browser` you're passing from the richurls.rb file? If so, maybe switch to kwargs? Also, a browser becomes a wrapper in this class?

Comment (grdw) [reply lib/url_fetcher.rb]: > I'm confused, is this just the default value for the browser you're passing from the richurls.rb file?

This is mainly for the specs. It's indeed the default from the `richurls.rb` file. 

The wrapper was the old name I used for the first iteration of this idea, renamed it correctly now 👍

Comment (grdw) [reply lib/url_fetcher.rb]: I also swapped to kwargs, looks way better in the specs 👯 . Thanks for the suggestion.

File: lib/url_fetcher.rb [hunk 2]
```diff
@@ -36,17 +42,10 @@ def digest
...
-      )
+      status, return_url, body = @browser.remote_call(@url)
 
-      response = session.get(@url)
-
-      if response.status < 400
-        decorated = BodyDecorator.decorate(
-          response.url, response.body, @attributes
-        )
+      if status < 400
```
Comment (lorenzograndi4) [inline lib/url_fetcher.rb:47]: Nit; magic number

**Pull request merged**